packages/nuxt/src/pages/runtime/page.ts

```diff
@@ -66,6 +66,9 @@ export default defineComponent({
       nuxtApp._isNuxtPageUsed = true
     }
     let pageLoadingEndHookAlreadyCalled = false
+
+    const routerProviderLookup: Record<string, ReturnType<typeof defineRouteProvider> | undefined> = {}
```

Comment:
it might be worth investigating whether a WeakMap would also resolve the issues linked above

(it might also be that the overhead for multiple RouteProviders is small enough not to matter, but it's worth verifying that pages/routes can still be garbage collected)

Reply:
@danielroe I've updated it to a version using `WeakMap`. If `routeProps.Component.type` can be garbage collected, then this approach should be more memory-efficient compared to using an object.